src/lib/FreeC/Frontend/Haskell/Simplifier.hs

```diff
@@ -480,11 +482,30 @@ simplifyTypeSchema (HSE.TyForall srcSpan (Just binds) Nothing typeExpr) = do
 -- Without explicit @forall@.
 simplifyTypeSchema typeExpr = do
   typeExpr' <- simplifyType typeExpr
-  let srcSpan  = IR.typeSrcSpan typeExpr'
-      typeArgs = map
-        (IR.TypeVarDecl NoSrcSpan . fromJust . IR.identFromQName)
-        (freeTypeVars typeExpr')
+  let srcSpan         = IR.typeSrcSpan typeExpr'
+      typeArgIdents   = freeTypeVars typeExpr'
+      typeArgSrcSpans = map (findTypeArgSrcSpan typeExpr') typeArgIdents
+      typeArgs        = zipWith IR.TypeVarDecl typeArgSrcSpans typeArgIdents
   return (IR.TypeSchema srcSpan typeArgs typeExpr')
+ where
+  -- | Finds the first occurrence of the type variable with the given name.
+  --
+  --   Returns 'NoSrcSpan' if 'findTypeArgSrcSpan'' returns @Nothing@.
+  --   Since the type arguments have been extracted using 'freeTypeVars',
+  --   'findTypeArgSrcSpan'' should never return @Nothing@.
+  findTypeArgSrcSpan :: IR.Type -> IR.TypeVarIdent -> SrcSpan
+  findTypeArgSrcSpan = fromMaybe NoSrcSpan .: flip findTypeArgSrcSpan'
+
+  -- | Like 'findTypeArgSrcSpan' but returns @Nothing@ if there is
+  --   no such type variable.
+  findTypeArgSrcSpan' :: IR.TypeVarIdent -> IR.Type -> Maybe SrcSpan
+  findTypeArgSrcSpan' = fmap IR.typeSrcSpan .: findFirstSubterm . isTypeVar
+
+  -- | Tests whether the given type is the type variable with the given name.
+  isTypeVar :: IR.TypeVarIdent -> IR.Type -> Bool
+  isTypeVar typeVarIdent (IR.TypeVar _ typeVarIdent') =
+    typeVarIdent == typeVarIdent'
+  isTypeVar _ _ = False
```

Comment:
Brittany does not want these function rules to be aligned, huh?

Reply:
As far as I know, Brittany does not align rules if one of them spans multiple lines.